### ISSUE TITLE
Create extractor for environmental logger

### ISSUE THREAD

**Issue by dlebauer** [terraref/computing-pipeline#115]
The file  [EnvironmentalLoggerAnalyser.py](https://github.com/terraref/computing-pipeline/blob/master/scripts/hyperspectral/EnvironmentalLoggerAnalyser.py) converts the environmental logger stream to a netCDF file

We should add this to the pipeline as an extractor by wrapping its execution in a pyClowder script

@FlyingWithJerome please ask @max-zilla if you need help getting started with this.

**Comment by FlyingWithJerome** [comment]
@dlebauer Got it, and sorry for the late reply.
But I need to update the EnvironmentalLoggerAnalyser.py to let it fit the newest format of the environmental logger data before add it to the pipeline.

**Comment by rachelshekar** [comment]
@FlyingWithJerome please update this issue

**Comment by rachelshekar** [comment]
@FlyingWithJerome - please update this issue

**Comment by FlyingWithJerome** [comment]
@rachelshekar Sorry, I think right now @max-zilla and @Zodiase are working on the extractor. I'm maintaining the source code of hyperspectral workflow.

**Comment by max-zilla** [comment]
@FlyingWithJerome is this the same extractor as https://github.com/terraref/computing-pipeline/issues/81?

That's the main one I think @Zodiase is working on. Unless @yanliu-chn has updates I haven't heard about, haven't done anything with this. I can get it done if no one else has started.

**Comment by dlebauer** [comment]
The environmental data extractor should be separate from the hyperspectral
extractor although the environmental data is required by the hyperspectral
data

**Comment by max-zilla** [comment]
@FlyingWithJerome is going to update EnvironmentalLogger.py file and then I will help create extractor.

**Comment by FlyingWithJerome** [comment]
Oh, I misunderstood; so the EnvironmentalLogger needs a separate extractor? 
Yes. I will keep updating the EnvironmentalLogger.py and create the extractor for it.
@max-zilla Maybe no. I think Dr. LeBauer @dlebauer  means EnvironmentalLogger.py works in Hyperspectral workflow, but needs an extra extractor

**Comment by dlebauer** [comment]
Correct, the environmental logger extractor needs to work independently
from the hyperspectral extractor

**Comment by FlyingWithJerome** [comment]
@Zodiase Please note that there are two versions of EnvironmentalLoggerAnalyser.py. They work for environmentallogger.json generated before and after a certain day; details were in #122 .

**Comment by Zodiase** [comment]
@dlebauer Should this one take priority over the other issues?

**Comment by dlebauer** [comment]
@Zodiase If the output of this extractor is required for #81, then it should be done first or concurrently.

**Comment by max-zilla** [comment]
@Zodiase I will begin working on this extractor while you focus on hyperspectral. We can hand off at some point when it makes sense, but I'll keep you updated here. There was an existing branch that was about a month old that didn't appear to have anything done on it, so I created a new branch:
https://github.com/terraref/computing-pipeline/tree/environmentlogger_extractor

**Comment by dlebauer** [comment]
@max-zilla have you spoken with @robkooper about the extent to which we can make this compatible with the PEcAn/BrownDog met workflow? It is worth understanding how these will link together. I haven't seen the outputs of this extractor, but ultimately we want to provide met products in the PEcAn format. If this isn't already in that format, we can add the conversion to PEcAn-CF as a separate step (later, when we deal with other met extractors).

**Comment by max-zilla** [comment]
@FlyingWithJerome @Zodiase I have initial version of extractor working:
https://github.com/terraref/computing-pipeline/tree/environmentlogger_extractor/scripts/environmental/environmentlogger

- I **modified EnvironmentalLoggerAnalyser.py** very slightly. My script calls mainProgramTrigger() directly with input/output given, so it would crash when trying to reference sys.argv[1] in that method. Since you pass in the command line args from main(), I replaced the sys.argvs with the variables fileInputLocation and fileOutputLocation. No other changes.
- Created an **environmental** directory under /scripts and put extractor directory there, as discussed in https://github.com/terraref/computing-pipeline/issues/160. Eventually hyperspectral extractor should also go in there.
- Adds the .nc output file to the same dataset the _environmentlogger.json file is in.

@FlyingWithJerome the only files this extractor needs are the Analyser.py script and the CalculationWorks.py file. If I understand correctly, the hyperspectral extractor will not need these files specifically, but only the output to Analyser.py?

I have not looked closely at hyperspectral script, but does that need both the _environmentlogger.netCDF file _and_ files from other sensors? Trying to think through the chaining of these two extractors and how to trigger the hyperspectral extractor once this extractor is done - right now the _envlogger.json files are going into their own datasets separate from the SWIR/VNIR files. Will we need to match the EL data with the HS data for the HS extractor?

**Comment by max-zilla** [comment]
Other comment: if the hyperspectral extractor truly only depends on the output to this extractor, we should also remove the Analyser.py/CalculationWorks.py files from the /hyperspectral directory so there is only one copy to maintain.

**Comment by FlyingWithJerome** [comment]
@max-zilla The Hyperspectral workflow used to depend on EL scripts and need the data collected from them, but I just have a glance at @czender 's script and seems that he had removed the running dependency on EL. So you are correct; for now, I think we can safely move the EL scripts to a new folder. Hyperspectral workflow may still need outputs from EL, but we can find another way to merge their results.
By the way, thank you for your modification!

**Comment by czender** [comment]
HS calibration has been put on hold because of problems like non-overlapping spectral regions, shadowing,.... In the future we will use EL-measured flux in HS calibration.

**Comment by max-zilla** [comment]
I have created a separate repository for this extractor (and eventually other environmental extractors) now that it is nearly complete:
https://github.com/terraref/extractors-environmental 

It has the latest file updates and renamings. I am going to close this issue and make an issue focused on deploying it now.
